lib/Conversion/TorchToLinalg/TorchToLinalg.cpp

```diff
@@ -2685,6 +2696,58 @@ class ConvertAtenMaxPool2dOp : public OpConversionPattern<AtenMaxPool2dOp> {
 };
 } // namespace
 
+namespace {
+class ConvertAtenConstantPadNdOp : public OpConversionPattern<AtenConstantPadNdOp> {
+public:
+  using OpConversionPattern::OpConversionPattern;
+  LogicalResult
+  matchAndRewrite(AtenConstantPadNdOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    if (failed(verifyLinalgCompatibleTypes(op, rewriter)))
+      return failure();
+    Location loc = op->getLoc();
+    Value self = adaptor.self();
+    auto type = self.getType().cast<RankedTensorType>();
+    int64_t rank = type.getRank();
+
+    Type elementType = self.getType().cast<RankedTensorType>().getElementType();
+    if (!elementType.isa<mlir::FloatType>())
```

Comment:
This type limitation can be removed once the `arith::TruncFOp` below is changed to use `convertScalarToDtype`